client/state/atomic-transfer/use-atomic-transfer-query.ts

```diff
@@ -0,0 +1,70 @@
+import { useQuery } from '@tanstack/react-query';
+import wpcom from 'calypso/lib/wp';
+import { transferStates, TransferStates } from 'calypso/state/automated-transfer/constants';
+import { SiteSlug } from 'calypso/types';
+
+interface ErrorResponse {
+	code: 'no_transfer_record';
+}
+
+interface SuccessResponse {
+	status: TransferStates;
+}
+
+const fetchLatestAtomicTransfer = (
+	siteSlug: SiteSlug
+): Promise< ErrorResponse | SuccessResponse > =>
+	wpcom.req.get( {
+		path: `/sites/${ siteSlug }/atomic/transfers/latest`,
+		apiNamespace: 'wpcom/v2',
+	} );
+
+interface UseAtomicTransferQueryOptions {
+	refetchInterval?: number;
+}
+
+const endStates: TransferStates[] = [
+	transferStates.NONE,
+	transferStates.COMPLETE,
+	transferStates.COMPLETED,
+	transferStates.FAILURE,
+	transferStates.ERROR,
+	transferStates.REVERTED,
+];
+
+export function useAtomicTransferQueryQueryKey( siteSlug ) {
+	return [ 'sites', siteSlug, 'atomic', 'transfers', 'latest' ];
+}
+
+export const useAtomicTransferQuery = (
+	siteSlug: SiteSlug,
+	{ refetchInterval, enabled }: UseAtomicTransferQueryOptions
+) => {
+	const { data, failureReason } = useQuery( {
+		queryKey: useAtomicTransferQueryQueryKey( siteSlug ),
+		queryFn: () => fetchLatestAtomicTransfer( siteSlug ),
+		refetchInterval,
+		enabled,
+	} );
+
+	if ( ! data && ! failureReason ) {
+		return {
+			isTransferring: false,
+			transferStatus: undefined,
+		};
+	}
+
+	if ( failureReason ) {
+		// Will happen for new sites when no transfer exists: a 404 is returned
+		// by the API which becomes failureReason in React Query
+		return {
+			transferStatus: transferStates.NONE,
```

Comment:
I'm assuming the request returns a successful status even if there was an error during the transfer. Can we double-check it?

Reply:
The HTTP response code is `200`. However, the payload looks like this:

```json
{
    "body": {
        "code": "no_transfer_record",
        "message": "Transfer record not found for blog `227821875`",
        "data": {
            "status": 404
        }
    },
    "status": 404,
    "headers": {
        "Allow": "GET"
    }
}
```
And React Query parses it to:

```js
{
  "status": "loading",
  "fetchStatus": "fetching",
  "isLoading": true,
  "isSuccess": false,
  "isError": false,
  "isInitialLoading": true,
  "dataUpdatedAt": 0,
  "error": null,
  "errorUpdatedAt": 0,
  "failureCount": 3,
  "failureReason": {
    "path": "/sites/mk9287907.wordpress.com/atomic/transfers/latest",
    "apiNamespace": "wpcom/v2",
    "method": "GET",
    "apiVersion": "1.1",
    "query": "",
    "callback": "207c6f39-c612-448f-acde-7684177e1f48",
    "supports_args": true,
    "supports_error_obj": true,
    "supports_progress": true,
    "name": "NotFoundError",
    "statusCode": 404,
    "status": 404,
    "message": "Transfer record not found for blog `227821875`",
    "code": "no_transfer_record",
    "data": {
      "status": 404
    }
  },
  "errorUpdateCount": 0,
  "isFetched": false,
  "isFetchedAfterMount": false,
  "isFetching": true,
  "isRefetching": false,
  "isLoadingError": false,
  "isPaused": false,
  "isPlaceholderData": false,
  "isPreviousData": false,
  "isRefetchError": false,
  "isStale": true
}
```